### PR TITLE
fix-command-rename-bug

### PR DESCRIPTION
**Pull request by qiaozha**
fix issue #710

### CODE REVIEW

File: src/modifiers.ts [hunk 1]
```diff
@@ -92,190 +93,200 @@ function hasSpecialChars(str: string): boolean {
...
         this.codeModel = session.model;
         this.getAllCommandGroup();
     }
 
     getAllCommandGroup() {
-        this.allCommandGroups = [];
-        for (const group of values(this.codeModel.operationGroups)) {
-            this.allCommandGroups.push(group.language['az'].command);
-        }
+        this.allCommandGroups = {};
```
Comment (lirenhe) [inline src/modifiers.ts:104]: It looks the operation name  is different from what the operation is doing.

File: src/modifiers.ts [hunk 2]
```diff
@@ -92,190 +93,200 @@ function hasSpecialChars(str: string): boolean {
...
+        let groupIdx = -1;
```
Comment (lirenhe) [inline src/modifiers.ts:105]: can we start from groupIdx = 0 and put groupIdx ++ after setting the value?

File: src/modifiers.ts [hunk 3]
```diff
@@ -92,190 +93,200 @@ function hasSpecialChars(str: string): boolean {
...
+            this.allCommandGroups[operationGroup.language['az'].command] = groupIdx;
+        });
     }
 
     async process() {
         directives = AzConfiguration.getValue(CodeGenConstants.directive);
         const options = AzConfiguration.getValue(CodeGenConstants.az);
-        if (!isNullOrUndefined(directives)) {
-            for (const directive of directives.filter((each) => !each.transform)) {
+        if (isNullOrUndefined(directives)) {
```
Comment (lirenhe) [inline src/modifiers.ts:115]: Can we add return value for this function?

Comment (lirenhe) [reply src/modifiers.ts:115]: Do we have test case for this function?

File: src/modifiers.ts [hunk 4]
```diff
@@ -92,190 +93,200 @@ function hasSpecialChars(str: string): boolean {
...
+                        : undefined;
+                const commandReplacer =
+                    directive.set !== undefined ? directive.set.command : undefined;
+                const commandDescriptionReplacer =
+                    directive.set !== undefined ? directive.set['command-description'] : undefined;
+                const groupReplacer = directive.set !== undefined ? directive.set.group : undefined;
+                const groupDescriptionReplacer =
+                    directive.set !== undefined ? directive.set['group-description'] : undefined;
+                let groupIdx = -1;
+                this.codeModel.operationGroups.forEach((operationGroup) => {
```
Comment (lirenhe) [inline src/modifiers.ts:145]: can we split the detailed logic into different small functions?

File: src/modifiers.ts [hunk 5]
```diff
@@ -92,190 +93,200 @@ function hasSpecialChars(str: string): boolean {
...
+                                const oriName = operation.language['az'].name;
+                                oldCommandArr.pop();
+                                const oldGroup = oldCommandArr.join(' ');
+                                let commonIdx = newCommandArr.length - 2;
+                                let newGroup = newCommandArr
+                                    .slice(0, newCommandArr.length - 1)
+                                    .join(' ');
+                                const subCommandGroup = newGroup;
+                                let newAzName = newCommandArr.last;
+                                if (newAzName == 'list') {
```
Comment (lirenhe) [inline src/modifiers.ts:207]: ===

File: src/modifiers.ts [hunk 6]
```diff
@@ -92,190 +93,200 @@ function hasSpecialChars(str: string): boolean {
...
+                                    const groupName = newCommandArr
+                                        .slice(0, commonIdx + 1)
+                                        .join(' ');
+                                    const newIndex = this.allCommandGroups[groupName];
+                                    if (!isNullOrUndefined(newIndex)) {
+                                        newAzName = newCommandArr
+                                            .slice(commonIdx + 1, newCommandArr.length)
+                                            .join(' ');
+                                        newGroup = newCommandArr.slice(0, commonIdx + 1).join(' ');
+                                        if (groupIdx != newIndex) {
```
Comment (lirenhe) [inline src/modifiers.ts:220]: !==

File: src/modifiers.ts (hunk 1 above)
Comment (qiaozha) [reply src/modifiers.ts:119]: changed it into initAllCommandGroup

File: src/modifiers.ts (hunk 2 above)
Comment (qiaozha) [reply src/modifiers.ts]: changed.

File: src/modifiers.ts (hunk 3 above)
Comment (qiaozha) [reply src/modifiers.ts:262]: seperate it into three functions operationGroupModify operationModify and paramModify

File: src/modifiers.ts (hunk 4 above)
Comment (qiaozha) [reply src/modifiers.ts]: seperate it into three functions operationGroupModify operationModify and paramModify

File: src/modifiers.ts (hunk 3 above)
Comment (qiaozha) [reply src/modifiers.ts:262]: added. we have unittest for this modifier plugin https://github.com/Azure/autorest.az/blob/616138f2d86d34cb6238e43359cf97738fa2f1bb/test/test-modifier.ts

File: src/modifiers.ts (hunk 5 above)
Comment (qiaozha) [reply src/modifiers.ts]: changed.

File: src/modifiers.ts (hunk 6 above)
Comment (qiaozha) [reply src/modifiers.ts]: changed

File: src/modifiers.ts [hunk 7]
```diff
@@ -90,249 +96,216 @@ function hasSpecialChars(str: string): boolean {
...
+            parameter.language['az'].description =
+                paramDescriptionReplacer || parameter.language['az'].description;
+        }
     }
 
-    getAllCommandGroup() {
-        this.allCommandGroups = [];
-        for (const group of values(this.codeModel.operationGroups)) {
-            this.allCommandGroups.push(group.language['az'].command);
+    operationGroupModify(operationGroup: OperationGroup, directive: any, groupIdx: number): void {
```
Comment (lirenhe) [inline src/modifiers.ts:149]: modifyOperationGroup

Comment (qiaozha) [reply src/modifiers.ts]: changed, and I also changed operationModify => modifyOperation, paramModify => modifyParam

**Review by lirenhe — APPROVED** [review]

**Pull request merged**